src/ext/database_diagram.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2019 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Create a database diagram of Cylc's database, reading its schema, and adding
+relationships manually."""
+
+import sqlite3
+import tempfile
+from contextlib import closing
+
+import pygraphviz as pgv
+from cylc.flow.rundb import CylcSuiteDAO
+from eralchemy import render_er, main as eralchemy_main
+from eralchemy.main import all_to_intermediary
+from docutils.parsers.rst import directives
+from sphinx.ext.graphviz import Graphviz
+
+__version__ = '1.0.0'
+
+ONE_TO_ONE = "?--?"
+ONE_TO_MANY = "*--?"
+
+
+def get_relationships():
+    """These are the database relationships. * means 0..N, and ? {0, 1}.
+
+    Ideally we would have these relationships in the database model, but doing
+    that would require further tests to prevent regressions for users
+    after they update Cylc. So we are hard-coding the relationships here
+    for now.
+
+    Returns:
+        list: with relationships in the eralchemy markdown syntax.
+    """
+    # TODO: remove this once the relationships are in the DB, and then automate
```

Comment:
Is there an issue on Cylc Flow for this?

Reply:
I tried to add the relationships there, but doing that I had no idea whether it would break for users with existing databases.

I think it should work just fine, though it could slow down things in case SQLite starts doing some constraint checking maybe. The lowest risk option was to avoid touching the schema, and instead just keep the relationships in this script :disappointed_relieved:

Reply:
I would have though existing databases would be OK, else we could add an upgrader. Speed impact I'm not too sure about.

@matthewrmshin any reason we haven't added foreign keys into the DB yet?

Reply:
Happy to add foreign keys.

IIRC, @kinow already tried that, but there were too many test battery failures that were taking too much effort to resolve.

Reply:
Oh yeah, there were the functional tests failing too. That was a bit scary, even though I think it failed because it was comparing the output of the database structure. I can create a ticket for this TODO so that this doesn't slip, and we can try again later? At least once that's done, we have all the tooling here, and it should be easier to implement (and to undo in case something goes south)